detectors/aws/ecs/ecs.go

```diff
@@ -125,6 +128,26 @@ func (detector *resourceDetector) Detect(ctx context.Context) (*resource.Resourc
 			}
 		}
 
+		arnParts := strings.Split(taskMetadata.TaskARN, ":")
+		// a valid arn should have at least 6 parts
+		if len(arnParts) < 6 {
+			return empty, errCannotParseTaskArn
```

Comment:
`errCannotParseTaskArn` does not effectively communicate the error. It is known why the ARN parsing failed. Include the reason in the returned errors.

Reply:
Done in 052fb008